src/core/Form/Checkbox/Checkbox.md

```diff
@@ -1,88 +1,212 @@
+
+<Checkbox
+  checked={checked}
+  onClick={(newState) => {
+    setChecked(newState.checkboxState);
+  }}
+>
+  I wish to be contacted regularly
+</Checkbox>;
+```
+
+### Disabled
+
+You can disable the component with the `disabled` prop. A Checkbox could be disabled when the user has not yet performed some other required action.
+
+```js
+import { Checkbox } from 'suomifi-ui-components';
+
+<Checkbox disabled>I accept the terms & conditions</Checkbox>;
+```
+
+### Accessing value with ref
+
+The value of a Checkbox can be accessed using React ref.
+
+```js
+import { Checkbox } from 'suomifi-ui-components';
+import { useRef } from 'react';
+
+const checkboxRef = useRef();
+
+<Checkbox
+  ref={checkboxRef}
+  onClick={() => console.log(checkboxRef.current.checked)}
+>
+  I accept the terms & conditions
+</Checkbox>;
+```
+
+### Large checkbox
+
+Use `variant="large"` on smaller screen sizes.
+
 ```js
 import { Checkbox } from 'suomifi-ui-components';
+
+<Checkbox variant="large">I accept the terms & conditions</Checkbox>;
+```
+
+### Checkboxes in group
+
+Use the `<ChexboxGroup>` component to place multiple Checboxes into a group.
+
+The group can have a common `groupHintText`, and an individual Checkbox inside the group can have its own `hintText`
+
+If you only allow the user to select a single option, use the <a href="#/Components/RadioButton">RadioButton</a> component instead.
+
+```js
+import { Checkbox, CheckboxGroup } from 'suomifi-ui-components';
 import React from 'react';
 
-const exampleRef = React.createRef();
-
-const [checked, setChecked] = React.useState(false);
-<>
-  <Checkbox
-    defaultChecked
-    hintText="This is an example hint text"
-    onClick={(value) => {
-      console.log(value);
-    }}
-  >
-    Regular checkbox that is checked and has a hint text
-  </Checkbox>
+<CheckboxGroup
+  labelText="How do you wish to be contacted?"
+  groupHintText="You can choose more than one option"
+>
+  <Checkbox defaultChecked>By email</Checkbox>
 
-  <Checkbox
-    hintText="Example hint text"
-    status="error"
-    statusText="You need to accept the terms and conditions to continue"
-  >
-    Regular checkbox with a hint text and an error message
+  <Checkbox hintText="We will call during office hours">
+    By phone
   </Checkbox>
 
-  <Checkbox variant="large">Large checkbox</Checkbox>
-
-  <Checkbox
-    variant="large"
-    defaultChecked
-    hintText="Example hint text"
-    ref={exampleRef}
-    onClick={() => {
-      console.log(exampleRef.current);
-    }}
-  >
-    Checked large checkbox with a hint text and ref
-  </Checkbox>
+  <Checkbox>By personal visit</Checkbox>
+</CheckboxGroup>;
+```
+
+### Group error status
+
+The ChexboxGroup can have a `groupStatus` which is passed down to all children. An individual Checkbox inside the group can still have its own status which will override the group status.
+
+Always use a descriptive `groupStatusText` with the error status.
```

Comment:
Even though `groupStatus` is "passed down to all children", `groupStatusText` is not, and we should make this clear. Unless I'm missing something, it seems the status text is simply an aria-live area, but it is not programmatically connected to the fieldset or the individual inputs. Individual Checkbox status text only overrides the group status in the sense that they use "assertive" instead of "polite", but they could also very well override each other if more than one checkbox has an error. Using aria-live on inline error messages like this is potentially messy and not a pattern I think we should promote, but the document should accurately represent the current functionality. Here's a suggestion:

```suggestion
Always use a descriptive `groupStatusText` with the error status. It uses `aria-live="polite"`, and will be read automatically by screen readers. The `statusText` for individual Checkboxes uses `aria-live="assertive"` and could override the group status.
```